x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/saml/SamlRealmSettings.java

```diff
@@ -47,6 +47,18 @@ public class SamlRealmSettings {
         key -> Setting.timeSetting(key, TimeValue.timeValueHours(1), Setting.Property.NodeScope)
     );
 
+    public static final Setting.AffixSetting<TimeValue> IDP_METADATA_HTTP_MIN_REFRESH = Setting.affixKeySetting(
+        RealmSettings.realmSettingPrefix(TYPE),
+        IDP_METADATA_SETTING_PREFIX + "http.minimum_refresh",
+        key -> Setting.timeSetting(key, TimeValue.timeValueMinutes(5), Setting.Property.NodeScope)
```

Comment:
> should we also set a min value for this setting to prevent absurdly low values

We can. I'm not sure what would constitute absurdly low though. I guess "minutes" is sensible, "seconds" is acceptable and "milliseconds" is a problem. I'm not sure what the precise dividing line would be (my inclination is 5s)

> maybe we should we add setting validation that this value is < http.refresh rather than setting them equal at runtime, i.e. prefer strict validation vs. lenient behavior

I considered that, but it means either a potential breaking change or a slightly more complex behaviour.
If an admin has set `http.refresh` to _4 minutes_ and `http.minimum_refresh` defaults to _5 minutes_ then the introduction of this new setting would prevent their node from starting.

To avoid that we need to only apply the validation if `http.minimum_refresh` has an explicit value, and keep the current behaviour otherwise.
  
> should this be dynamic?

It would be helpful if `http.refresh` was dynamic, because if you knew that you had published updated (remote) metadata, you could set the `refresh` time down to force it to be reloaded more quickly. 
It's not so important to set `http.minimum_refresh` dynamically because of the automatic reload behaviour you mention. 

> should we deprecate the http.refresh in favor of http.max_refresh?

We could. It's arguably more consistent, but my gut is that it's one of the annoying changes that isn't worth it.

I don't think having `http.refresh` and `http.minimum_refresh` is terribly confusing, even if it isn't consistent - in fact I expect most admins who need to fiddle with refresh times should just set  `http.refresh` and ignore `http.minimum_refresh` anyway.

On balance, forcing every admin who has configured `http.refresh` to go through the deprecation process doesn't seem justified just to make things a bit more consistent - it doesn't feel like it meets the necessary threshold to force that pain onto users.

Reply:
> I'm not sure what the precise dividing line would be (my inclination is 5s)

I ended up setting the minimum to 500ms. Mostly because I had a test that relied on setting it to a very low value and making sure it did a refresh in the background. If the limit was 5s then the test would take >5s to run (for no great reason).

500ms still stops people setting it to tiny values, and if they really want it to refresh twice every second, I guess they can.